editor/src/core/shared/math-utils.ts

```diff
@@ -970,14 +970,16 @@ export function rectanglesEqual<C extends CoordinateMarker>(
 ): boolean {
   if (isInfinityRectangle(first) || isInfinityRectangle(second)) {
     return isInfinityRectangle(first) && isInfinityRectangle(second)
-  } else {
+  } else if (isFiniteRectangle(first) && isFiniteRectangle(second)) {
+    // TODO: why is this extra if necessary? They have to be both finite because of the else branch
```

Comment:
I wonder if this is because `||` can short circuit if `isInfinityRectangle(first)` returns `true`, which is not to say that's "correct" I just wonder if they've had to make this more dumb because of some edge case bug.

Reply:
I think type narrowing + short circuiting is the culprit too. The `if` with the `||` operator can succeed because of either of the `isInfinityRectangle` calls, so the types of `first` or `second` cannot be narrowed at compile time.

I'd restructure the function so that we fully narrow the types of `first` and `second` with an `isFiniteRectangle(first) && isFiniteRectangle(second)` check, and otherwise return `isInfinityRectangle(first) && isInfinityRectangle(second)`.

Reply:
Fixed in b49041dc766e24e3b51d6c4f2782417b1e211131